### PR TITLE
fix ci issues limo dataset example

### PR DESCRIPTION
**Pull request by JoseAlanis**
#### Reference issue
Fixes CircleCI issues in PR [#6113](https://github.com/mne-tools/mne-python/pull/6113)

#### What does this implement/fix?
set ``update_path=True`` as default in limo dataset fetcher.
modifies ``mne.datasets.utils._download_all_example_data`` to include limo dataset.

### CODE REVIEW

**Comment by agramfort** [comment]
Is it the same default as for all fetchers ? It should be

**Comment by JoseAlanis** [comment]
Its hard to say. As far as I can tell, some datasets have ``update_path=True`` (e.g., brainstrom, eegbci) and some have ``update_path=None`` (e.g., megsim, physionet) as default.

**Comment by agramfort** [comment]
Can we make this consistent but in a next PR?

**Comment by dengemann** [comment]
CIs green!

**Comment by JoseAlanis** [comment]
> Can we make this consistent but in a next PR?

sure, do we wanna let the user decide whether to update or should ``update_path=True`` be the default?

File: examples/datasets/plot_limo_data.py [hunk 1]
```diff
@@ -49,7 +49,7 @@
 print(__doc__)
 
 # fetch data from subject 2 and interpolate missing channels
-limo_epochs = limo.load_data(subject=2)
+limo_epochs = limo.load_data(subject=2, update_path=True)
```
Comment (agramfort) [inline examples/datasets/plot_limo_data.py]: update_path is not set explicitly in any example AFAIK

is this necessary here?

File: mne/datasets/utils.py [hunk 2]
```diff
@@ -595,6 +595,7 @@ def _download_all_example_data(verbose=True):
         fetch_hcp_mmp_parcellation()
     finally:
         sys.argv.pop(-1)
+    limo.load_data(subject=2, update_path=True)
```
Comment (agramfort) [inline mne/datasets/utils.py]:
```suggestion
    limo.load_data(subject=2)
```

File: examples/datasets/plot_limo_data.py (hunk 1 above)
Comment (JoseAlanis) [reply examples/datasets/plot_limo_data.py]: no not really. The update is done automatically. I'll delete it from the files calling ``limo.load_data``

**Comment by agramfort** [comment]
I am not I get the fix here. The logic is the following. It's None by default so user
gets asked when the config variable would be changed by the download. So if you
call the load_data with update_path=True the first time than any time you call it afterwards
should work without any prompt. As circle calls it with update_path=True why is it failing?
what do we fix here?

**Comment by JoseAlanis** [comment]
Sorry, I probably should have explained the issue a little bit better.
The original default setting in the referenced PR was ``mne.datasets.limo.load_data(update_path=None)``.  The function was called without setting ``update_path=True`` explicitly in the example.  If I understand correctly, this is probably what caused CircleCI failure.

The suggestion was to change ``update_path`` to be consistent with what’s done for the brainstorm dataset. i.e., set ``update_path=True`` as default.

In the first commit I had added ``update_path=True`` to the example (to get rid of the CI error). However, this seemed unnecessary after changing ``update_path=True`` to be default , so I deleted the arguments from the example and tests in a second commit.

Alternatively, we could just leave ``update_path=None`` as default and call the function with ``update_path=True`` in the example and test. But this probably raises the question of what we want to be the consistent default path option for all datasets.

Any suggestions?

**Comment by larsoner** [comment]
>  The function was called without setting update_path=True explicitly in the example. If I understand correctly, this is probably what caused CircleCI failure.

What caused the failure was this in combination with not having a call to `data_path(..., update_path=True)` in `mne.datasets.limo.data_path`.

> Alternatively, we could just leave update_path=None as default and call the function with update_path=True in the example and test. But this probably raises the question of what we want to be the consistent default path option for all datasets.

Looking deeper it looks like the pattern is that:

- downloaders for datasets that are a single file / archive use a `data_path` function with `update_path=True` by default, and in `_download_all_example_data` get called without passing anything to `update_path` 
- downloaders that fetch individual files  use a `load_data` function with `update_path=None` by default, and in `_download_all_example_data` get called with `update_path=True`

Not sure if this was done intentionally or not. So since this is a single fetcher using `load_data`, I guess the consistent thing would be to use `update_path=None` as a default and pass `update_path=True` in `_download_all_example_data`. Sorry for the misguidance to mimic `brainstorm`, I didn't realize we did these things differently!

**Comment by larsoner** [comment]
(And if we want, in a follow-up PR at some point we can change all the `data_path` args to be `None`. Don't want to do it here in case it breaks something and holds this up more.)

**Comment by agramfort** [comment]
> I guess the consistent thing would be to use update_path=None as a default and pass update_path=True in _download_all_example_data.

+1

**Comment by JoseAlanis** [comment]
>  I guess the consistent thing would be to use `update_path=None` as a default and pass `update_path=True` in `_download_all_example_data`.

Alright, no worries. I can take care of that. So just for clarification, this would mean that  `update_path=True` should also be passed to to `limo.load_data` in both  `_download_all_example_data` and the example file, right?

**Comment by agramfort** [comment]
it should not be needed for examples as it will be called after
_download_all_example_data

>

**Comment by JoseAlanis** [comment]
Alright, thanks. I changed the code according the suggestions. Hope it makes CircleCi happy.

**Comment by agramfort** [comment]
LGTM but now we have an unrelated CI failure... same as in other PRs...

**Comment by larsoner** [comment]
Let's at least get circle green on master. I can fix pytest tomorrow

**Pull request merged**